cmake/toolchain.cmake

```diff
@@ -1,4 +1,5 @@
 # Require C++11.
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
-set(CMAKE_CXX_EXTENSIONS Off)
\ No newline at end of file
+set(CMAKE_CXX_EXTENSIONS Off)
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```

Comment:
Why I don't understand why it is needed here. Shouldn't it be only enabled in Hera?

Reply:
The dependencies are built statically without `-fPIC`, so when hera is built as a shared library and is linked against for example `libboost_system-mt.a`, the linker fails with:

`relocation R_X86_64_32S against '.rodata' can not be used when making a shared object; `

Reply:
I've submitted https://github.com/ewasm/hera/pull/162 instead which adds `CMAKE_POSITION_INDEPENDENT_CODE` to Hera only.